packages/ckeditor5-font/src/ui/colorui.ts

```diff
@@ -84,6 +87,7 @@ export default class ColorUI extends Plugin {
 	) {
 		super( editor );
 
+		this.undoStepBatch = editor.model.createBatch();
```

Comment:
There are 4 places where you create a new batch:

1.  Here, when ColorUI plugin is initialised.
2.  On Save button execution.
3.  On Cancel button execution.
4.  On dropdown closure.

Correct me if I'm wrong, but instead of all that, we could reset a batch just whenever a dropdown is being open (option A), or even only whenever the color picker view is shown (option B). That's much easier to maintain and less error-prone.

To implement option A, we need to create a batch in this tree:

```
dropdownView.on( 'change:isOpen', ( evt, name, isVisible ) => {
				if ( !dropdownContentRendered ) {
					dropdownContentRendered = true;
					dropdownView.colorTableView!.appendUI();
				}

				if ( isVisible ) {
				[create batch here...]
```

And then we can remove all the other calls.

Option B is a bit more complex as we'd need to create a new event (called e.g. `showColorPicker`), fire it by `colorGridsPageView`, delegate it to `ColorTableView`, and then in `ColorUI` plugin listen on `showColorPicker` event fired by `colorTableView` and only then reset the batch.

Reply:
`colorGridsPageView ` doesnt know nothing about when color picker is shown, but we already have in `colorTableView ` method `showColorPicker ` which is called when we need to show color picker so there is no need to delegate it.

Reply:
We discussed this and though it admittedly works, it cuts some corners and may be limiting the granularity, so we'll stay with the `colorGridsPageView` being the original emitter.